### PR TITLE
Add Rename, Rare and sort commands spl docs

### PR DESCRIPTION
**Pull request by venkateshamatam**

### CODE REVIEW

File: docs/spl-docs/sort-command.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+**Default:** 10000
+
+#### desc
+**Syntax:** `d | desc`\
+**Description:** Inverts the sequence of the results. When multiple fields are designated, it inversely arranges the values within those fields according to their specified sequence. For instance, with three fields indicated, the `desc` parameter inversely organizes the values in the first field. For any group of identical values in the first field, it similarly inverts the order of related values in the second field, and for each group of matching values in the second field, it reverses the order of associated values in the third field.
+
+### Sort Field Options
+
+#### \<sort-field\>
+**Syntax:** `<field> | auto(<field>) | str(<field>) | ip(<field>) | num(<field>)`\
```
Comment (AndrewHess) [inline docs/spl-docs/sort-command.md:39]: nit: change `|` to "or". See https://github.com/siglens/siglens-docs/pull/95#discussion_r1674394930

Same comment some other places

File: docs/spl-docs/sort-command.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+1. **Use the sort field options to specify field types**
+Sort the results by the `http_status` field in ascending order, and then sort by `latency` in descending order to see the slowest responses at the top for each status code.
+
+    ```spl
+    ... | sort num(http_status), -num(latency)
+    ```
+
+2. **Specifying the number of results to sort**
+Sort first 100 results in descending order of the `latency` field and then by the `city` value in ascending order. This example specifies the type of data in each of the fields. The `latency` field contains numbers and the `city` field contains strings.
```
Comment (AndrewHess) [inline docs/spl-docs/sort-command.md:72]: I think you meant for the description of the example to render on the line after the title of the example. If so, you need to add another empty line (e.g., after line 64)
image

File: docs/spl-docs/rare-command.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
+
+# rare Command
+
+## Description
+
+Displays the least common values within a specified field.
+
+Identifies the rarest combination of values across all fields listed. When a `<by-clause>` is provided, it yields the least common combinations of values for each unique combination of the group-by fields.
+
+This command functions similarly to the top command but focuses on identifying the least common values instead of the most common ones.
```
Comment (AndrewHess) [inline docs/spl-docs/rare-command.md:10]: > ... to the top command ...

nit: use inline code for `top`

File: docs/spl-docs/sort-command.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+**sort**
+[\<count\>]
+**\<sort-by-clause\>...**
+[desc]
+
+### Required Arguments
+
+#### \<sort-by-clause\>
+**Syntax:** `[ - | + ] <sort-field>, ( - | + ) <sort-field> ...`\
```
[the remark this answers is not in the document]
Comment (venkateshamatam) [reply docs/spl-docs/sort-command.md:21]: The first `<sort-field>` is required but the `+`/`-` is optional(it assumes sort by ascending if the `+`/ `-` is not specified) for all arguments, and all `<sort-field>` after the first  one are optional. Splunk documentation mentions this in the same format as above - Should I update it to use `()` or `[]` for all of them? 


https://docs.splunk.com/Documentation/SplunkCloud/9.2.2403/SearchReference/Sort

File: docs/spl-docs/sort-command.md [hunk 5]
```diff
@@ -62,31 +61,38 @@ The required syntax is in bold
...
 3. **Specifying descending and ascending sort orders**
+
 To analyze user engagement by location and performance, sort results by the "country" field in ascending order to group data by country, and then by the "latency" field in descending order to see the slowest responses at the top within each country.
 
 	```spl
 	... | sort country, -latency
 	```
 
-5. **Fetching the Latest Event**: Retrieve the most recent event by sorting in descending order based on the `timestamp` field.
+5. **Fetching the Latest Event**
```
Comment (AndrewHess) [inline docs/spl-docs/sort-command.md:86]: You skip number 4

File: docs/spl-docs/sort-command.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndrewHess) [reply docs/spl-docs/sort-command.md:20]: Ok. What you have is good then

**Review by AndrewHess — COMMENTED** [review]
Also change this PR to merge into `develop` instead of into `spl-docs` (I gave the same comment with a little more detail here: https://github.com/siglens/siglens-docs/pull/94#pullrequestreview-2178731909)

File: docs/spl-docs/sort-command.md (hunk 5 above)
Comment (venkateshamatam) [reply docs/spl-docs/sort-command.md]: Fixed this in the latest commit

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**